src/rcc.rs

```diff
@@ -133,25 +224,79 @@ impl CFGR {
             })
         }
 
+        // Set up rcc based on above calculated configuration.
         let rcc = unsafe { &*RCC::ptr() };
-        if let Some(pllmul_bits) = pllmul_bits {
-            // use PLL as source
 
+       
+        // Enable requested clock sources
+        // HSI
+        if self.enable_hsi.unwrap_or(false) {
+            rcc.cr.write(|w| w.hsion().set_bit());
+            while rcc.cr.read().hsirdy().bit_is_clear() { }
+        }
+        // HSI14
+        if self.enable_hsi14.unwrap_or(false) {
+            rcc.cr2.modify(|_, w| w.hsi14on().set_bit());
+            while rcc.cr2.read().hsi14rdy().bit_is_clear() {
+                // nothing
```

Comment:
These empty loops aren't consistent. I'd do it without the comment inside

Reply:
Ok cool.